addons/custom-block-shape/userscript.js

```diff
@@ -0,0 +1,247 @@
+export default async function ({ addon, global, console }) {
+  var multiplier = addon.settings.get("paddingSize") / 100;
+  var BlocklyInstance = await addon.tab.traps.getBlockly();
+
+  (function (Blockly) {
+    const workspace = Blockly.getMainWorkspace();
```

Comment:
you should be using Blockly.getMainWorkspace() every time you need the workspace as it will switch whenever the user enters or leaves the editor

Reply:
I'll change it, but don't JS constants store pointers to the value rather than actually freezing the value?

Reply:
its confusing

it stores a pointer only for objects (including arrays, dom, classes, etc). Other types it stores the actual value.

But I think since `Blockly.getMainWorkspace()` is a function, the value is not stored anywhere yet so it has to make a new pointer.

I think